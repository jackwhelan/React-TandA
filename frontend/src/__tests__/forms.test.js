import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';

import LoginForm from '../components/forms/LoginForm.component';
import ModifyUserForm from '../components/forms/ModifyUserForm.component';
import RegistrationForm from '../components/forms/RegistrationForm.component';

describe('Form Components', () => {
    test('Login', () => {
        const component = renderer.create(<LoginForm/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Modify User', () => {
        const component = renderer.create(<Router><ModifyUserForm /></Router>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Register', () => {
        const component = renderer.create(<RegistrationForm />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})