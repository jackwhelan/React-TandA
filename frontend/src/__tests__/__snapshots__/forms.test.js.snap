// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Components Login 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="mt-4"
  >
    <label>
      Username
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="mt-4 mb-5"
  >
    <label>
      Password
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-black"
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`Form Components Modify User 1`] = `null`;

exports[`Form Components Register 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="mt-4"
  >
    <label>
      First Name
    </label>
    <input
      className="form-control"
      maxLength="30"
      minLength="2"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="mt-4"
  >
    <label>
      Last Name
    </label>
    <input
      className="form-control"
      maxLength="30"
      minLength="2"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="mt-4"
  >
    <label>
      Username
    </label>
    <input
      className="form-control"
      maxLength="30"
      minLength="3"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="mt-4"
  >
    <label>
      Password
    </label>
    <input
      className="form-control"
      maxLength="100"
      minLength="6"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="mt-4 mb-5"
  >
    <label>
      Email
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
  </div>
  <button
    className="btn btn-black"
    type="submit"
  >
    Submit
  </button>
</form>
`;
